File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-3.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int left = 0;
        int right = 0;
        int maxFreq = 1;
        int res = 0;
        unordered_map<char, int> counts;
        
        for (int right = 0; right < s.size(); right++) {
            counts[s[right]]++;
            maxFreq = max(counts[s[right]], maxFreq);

            while ((right - left + 1 - maxFreq) > k) {
                counts[s[left]]--;
                left++;
            }
            res = max(res, right - left + 1);
        }
        return res;
    }
};

File: Data Structures & Algorithms/house-robber/submission-3.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]
        dp = [0] * len(nums)
        for i in range(len(nums)):
            dp[i] = max(nums[i] + dp[i - 2], dp[i - 1])
        return max(dp)